icu4c/source/test/intltest/numbertest_parse.cpp

```diff
@@ -408,5 +409,83 @@ void NumberParserTest::testInfiniteRecursion() {
     assertEquals("Unlimited recursion, expected double", -5.0, resultObject.getDouble(status));
 }
 
+void NumberParserTest::test13733_StrictAndLenient() {
+    IcuTestErrorCode status(*this, "test13733_StrictAndLenient");
+
+    static const struct TestCase {
+        const char16_t* inputString;
+        const char16_t* patternString;
+        long expectedStrictParse;
```

Comment:
Here and elsewhere: Don't use `long` in ICU4C. Use a sized type like `int64_t` instead.

Reply:
Changed to `int64_t`